components/kserve/kserve.go

```diff
@@ -149,6 +153,11 @@ func (k *Kserve) ReconcileComponent(ctx context.Context, cli client.Client, resC
 			return err
 		}
 	}
+
+	if err := k.setupKserveConfigAndDependencies(ctx, cli, dscispec); err != nil {
```

Comment:
This moved some logic outside the `if !enabled` block.

This leads to:
* trying to configuring Serverless, and
* trying to apply the patch to configure the default deployment mode

... regardless of KServe management state, which does not looks right.

Reply:
ah, missed this. Fixed by only calling the function if kserve is enabled